File: google/main/src/core/query/Query.java
package main.src.core.query;


public class Query {

    public AndTerms andTerms;
    public OrTerms orTerms;
    public ExcTerms excTerms;
    
    public Query(String queryString) {
        this.collectTerms(queryString);
    }

    private void collectTerms(String expression) {
        this.andTerms = new AndTerms(expression);
        this.orTerms = new OrTerms(expression);
        this.excTerms = new ExcTerms(expression);
    }

    @Override
    public String toString() {
        return  "Query\n" + 
                "ANDs : " + this.andTerms.toString() + "\n" + 
                "ORs  : " + this.orTerms.toString() + "\n" + 
                "EXCs : " + this.excTerms.toString();
    }
}

File: Bagherzadeh/BagherEngine/QuerySystem/Query.cs
namespace BagherEngine.QuerySystem
{
    public class Query
    {
        public AndTerms Ands { get; set; }
        public OrTerms Ors { get; set; }
        public ExcTerms Excs { get; set; }

        public Query(string queryString)
        {
            CollectTerms(queryString);
        }

        private void CollectTerms(string expression)
        {
            Ands = new AndTerms(expression);
            Ors = new OrTerms(expression);
            Excs = new ExcTerms(expression);
        }
    }
}
